### ISSUE TITLE
Data should be projected in a metric system

### ISSUE THREAD

**Issue by francis33** [zoran-cuckovic/QGIS-terrain-shading#7]
Hello,

I would like to understand the message "ERROR ! Raster data should be projected in a metric system ! "
I noticed on your website that " data should be projected in a metric system."

I use a DEM file and the WGS84 projection system with QGIS.
Is WGS84 a wrong projection system to use the plugin ?
Is the error message linked with this projection system ?

How can i correct my mistake ?

Here's the full error message from QGIS :

Version de QGIS : 3.14.1-Pi
Révision du code : de08d6b71d
Version de Qt : 5.11.2
Version de GDAL : 3.0.4
Version de GEOS : 3.8.1-CAPI-1.13.3
Version de Proj : Rel. 6.3.2, May 1st, 2020
Traitement de l'algorithme…
Démarrage de l'algorithme 'Shadow depth'…
Paramètres en entrée:
{ 'ANGLE' : 10, 'DIRECTION' : 315, 'INPUT' : 'C:/Users/Cascade Interchange/Documents/00. Centre de données/10. QGIS Projects/02. Northland Interchange/20. Projects/W118 WGS84.tif', 'OUTPUT' : 'TEMPORARY_OUTPUT', 'SMOOTH' : True }

****** 
ERROR! 
Raster data should be projected in a metric system!
Échec d'exécution au bout de 0.19 secondes

Thanks a lot for your help !
Best
Francis

**Comment by zoran-cuckovic** [comment]
Hello,
WGS84 is not a true geographic projection : see my [blog post ](https://landscapearchaeology.org/2020/wgs/). Your data is in degrees, not in metres.

**Comment by francis33** [comment]
Hi,

Thanks a lot for your quick reply.
I learn a lot with your blog post. Following that, I read a lot of articles concerning projection systems.

I used the projection system "EPSG:3857 - WGS 84 / Pseudo-Mercator - Projeté".
My data are now in meters and the plugin worked. But i had the message : 

****** 
ERROR! 
Raster pixels are irregular in shape (probably due to incorrect projection)!

Having a look to the ouput file, I noticed a sort of 'grid' in the visual result (see attached file named "grid"). 
Is the choice of "EPSG:3857 - WGS 84 / Pseudo-Mercator - Projeté" a bad choice ?
Is this "bad" result linked to the error message ?

Is a UTM projection system a best choice ? 
(I work on north american areas, including canada. I'm studying global relief) 

Thanks a lot for your help.
Francis

![Ambiant Occlusion](https://user-images.githubusercontent.com/53442186/89614182-341fbf80-d884-11ea-9c05-e6934f2588c5.jpg)
![grid](https://user-images.githubusercontent.com/53442186/89614194-37b34680-d884-11ea-87c2-064980386aa0.jpg)
